### ISSUE TITLE
Properly support ModulePartVariants

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/TweakScale#13]
Tonka Crash [nailed the reason](https://forum.kerbalspaceprogram.com/index.php?/topic/179030-141151-tweakscale-under-new-management-2018-1127/&do=findComment&comment=3502250) for at least some parts ending up with negative mass (if everyone of that is due this, is something to be researched yet).

On MH, the  The T-12 tube (tube1.cfg) has negative baseMass and then positive mass on the variants - by inference, these masses need to be scaled too.

Task: properly implement support for ModulePartVariants

**Comment by Lisias** [comment]
TweakScale 2.4.0.7 will "solve" this issue by dropping support for parts using ModulePartVariants. Sorry.

Proper support will be implemented in the new code tree.

**Comment by snakeru** [comment]
Hi @Lisias 
According to your request, posing a KSP.log (same log posted on issue #11 ).

**Comment by orchidalloy** [comment]
Hey, this issue causes the mod to show a warning **every single time the game opens** for **vanilla parts** (structural tubes and engine plates). Please help me disable the warning completely, it's really stupid.

**Comment by Lisias** [comment]
> Hey, this causes the mod to show a warning **every single time the game opens** for **vanilla parts** (structural tubes and engine plates). I don't care that the mod thinks they can't be resized. Please help me disable the warning completely, it's really stupid.

It's a warning saying to you that some parts will not be scalable, as TweakScale doesn't knows (yet) how to cope with Variants with Mass. This is scheduled to be tackled down in the next few days and will be fixed on TweakScale 2.4.4.0 . Can you wait some time?

**Comment by orchidalloy** [comment]
Oh, if you're fixing it that's awesome. I just wanted to disable the warning message from appearing.

**Comment by Lisias** [comment]
> Oh, if you're fixing it that's awesome. I just wanted to disable the warning message from appearing.

I just remembered that someone had already complained about it on Forum, and someone else (zer0Kerbal IIRC) had published something about. (I was on working time when I replied you).

[Download this file](https://github.com/net-lisias-ksp/TweakScale/blob/master/Extras/TweakScale/Workarounds/VariantsWithMass.cfg) and put it somewhere on the GameData (I suggest `GameData/__LOCAL/TweakScale/WorkArounds`) and it will remove the TweakScale module from the affected parts before TweakScale tries to inspect it.

Just remember to delete it when TweakScale 2.4.4.0 is released, otherwise that parts will not be scalable even after I implement the proper support.

My apologies for the time I'm taking to implement this thing - TweakScale 2.4.4 was intended to be issued on February or March, but KSP 1.9.x came broking some serious thingies (see [KSP-Recall](https://forum.kerbalspaceprogram.com/index.php?/topic/192048-*)) and then COVID19 happened and, boy... **everything** went upside down around here...
